src/features/account/components/modals/AddToUserCircleModal.tsx

```diff
@@ -26,50 +28,75 @@ export const AddToUserCircleModal: FunctionComponent<
   );
   const dispatch = useDispatch();
 
-  const {
-    isLoading,
-    isVerified,
-    payload,
-    submitTypedSignRequest,
-    signatureError,
-  } = useTypedSignTrustCredential();
+  const [isLoading, setIsLoading] = useState(false);
+  const { issuerAddress, signMessage, signError, accountVCBuilder } =
+    useVerifiableCredential();
 
-  const { showErrorMsg, showSuccessMsg } = useToastMsg();
+  const { showSuccessMsg, showErrorMsg } = useToastMsg();
 
   const shortSubAddress = useMemo(
     () => trimAddress(subjectAddress),
     [subjectAddress],
   );
 
-  useEffect(() => {
-    if (signatureError) {
-      if (signatureError.type === SignatureErrorTypes.Error) {
-        showErrorMsg({
-          title: t`Failed to verify signature`,
-          description: signatureError.message,
-        });
-      } else {
+  const onSignButtonClick = useCallback(() => {
+    if (!issuerAddress) {
+      return;
+    }
+    if (!isAddress(subjectAddress)) {
+      return;
+    }
```

Comment:
```suggestion
    if (!issuerAddress || isAddress(subjectAddress)) {
      return;
    }
```

Reply:
actually you dont need to verify issuerAddress too, issuerAddress already checked in underlying

Reply:
Sorry, need to revert your suggestion above due to IDE type error, for subjectAddress will be null.

Reply:
> 

that one is not fixed

Reply:
```
if (!issuerAddress || isAddress(subjectAddress)) {
      return;
    }
```

Reply:
as discussed, this suggestion will have IDE type warning, i have reverted the change to split the if.